platform/mellanox/mlnx-platform-api/sonic_platform/component.py

```diff
@@ -628,7 +635,7 @@ def get_firmware_update_notification(self, image_path):
         return notification
 
     def install_firmware(self, image_path):
-        return self.__install_firmware(image_path)
+        return self.__install_firmware(image_path, allow_reboot=False)
```

Comment:
@alexrallen this will break the original `non-autoupdate` behaviour

Reply:
Lets merge this discussion with your comment in the upgrade script.